src/sentry/auth/access.py

```diff
@@ -47,68 +54,141 @@ def get_permissions_for_user(user_id: int) -> FrozenSet[str]:
     return cast(FrozenSet[str], union)
 
 
-@dataclass(frozen=True, eq=True)
-class SsoState:
-    is_required: bool
-    is_valid: bool
+class Access(abc.ABC):
```

Comment:
Access now becomes an implementation-less interface.

There are two base classes that derive from this interface, one providing a purely data driven access and the other, the existing DB driven access.